libcalico-go/lib/clientv3/tier.go

```diff
@@ -46,6 +46,17 @@ type tiers struct {
 // Create takes the representation of a Tier and creates it.  Returns the stored
 // representation of the Tier, and an error, if there is any.
 func (r tiers) Create(ctx context.Context, res *apiv3.Tier, opts options.SetOptions) (*apiv3.Tier, error) {
+	if res != nil {
+		// Since we're about to default some fields, take a (shallow) copy of the input data
```

Comment:
I think we should make a common "defaulting" function to ensure the logic is the same for both.